pubsub/drivertest/drivertest.go

```diff
@@ -236,8 +236,13 @@ func testSendReceive(t *testing.T, newHarness HarnessMaker) {
 	}
 	defer cleanup()
 
-	want := publishN(ctx, t, top, 3)
-	got := receiveN(ctx, t, sub, len(want))
+	wantChan := make(chan []*pubsub.Message)
+	gotChan := make(chan []*pubsub.Message)
+	n := 3
+	go func() { gotChan <- receiveN(ctx, t, sub, n) }()
+	go func() { wantChan <- publishN(ctx, t, top, n) }()
```

Comment:
You say on the issue that something has to be receiving while the messages are being published. But here it's possible that the publish goroutine starts, and maybe even completes, before Receive is ever called.

Reply:
Fair point. Anyway, I'm removing this PR since we agreed that we're not accepting the NATS pubsub driver for now and more likely supporting it in a different API.